### PR TITLE
Various CSS stuff

### LINKED ISSUE #14420: `Bun.build` with `experimentalCss` doesn't work with nested CSS
### What version of Bun is running?

1.1.30+7996d06b8

### What platform is your computer?

Darwin 24.0.0 arm64 arm

### What steps can reproduce the bug?

If you have a CSS file like:

```css
/* style.css */
body {
	font-family: ui-sans-serif, sans-serif;
	background-color: black;
	display: grid;
	width: 100dvw;
	height: 100dvh;
	margin: 0;
	place-items: center;
}

h1 {
	color: white;
}
```

then it builds correctly. However, if the CSS is nested, like:

```css
/* style-breaking.css */
body {
	font-family: ui-sans-serif, sans-serif;
	background-color: black;
	display: grid;
	width: 100dvw;
	height: 100dvh;
	margin: 0;
	place-items: center;

	h1 {
		color: white;
	}
}
```

then the bundler breaks.

Here's the bundler code I am using:

```ts
Bun.build({
	entrypoints: ["style.css"], // This one works fine
	outdir: "dist",
	experimentalCss: true,
});

Bun.build({
	entrypoints: ["style-breaking.css"], // This one breaks
	outdir: "dist",
	experimentalCss: true,
});
```

### What is the expected behavior?

Both CSS files should be built correctly. CSS nesting has been [supported in all major browsers](https://caniuse.com/css-nesting) for some time now.

### What do you see instead?

The breaking stylesheet fails to build, and I get an error in my console:

```
zsh: illegal hardware instruction  bun run build.ts
```

and with the CLI:

```
zsh: illegal hardware instruction  bun build --experimental-css style-breaking.css --outdir dist
```

### Additional information

_No response_

### LINKED ISSUE #14487: Bun crashes when building empty css file with --experimentalCss
### How can we reproduce the crash?

```tsx
//Component.tsx

import React from 'react'
import './styles.css'

export default function Whatever() {
  return <div>dsfsdf</div>
}
```

The `styles.css` file exists but it's empty

```ts
//build.ts

await Bun.build({
  entrypoints: ['Component.tsx'],
  outdir: "out",
  experimentalCss: true,
})
```

and doing `bun run build.ts` gives the seg fault

it works fine if `styles.css` has anything (even comments)

### Relevant log output

panic: Segmentation fault at address 0x8
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

### Stack Trace (bun.report)

Bun v1.1.30 ([`7996d06`](<https://github.com/oven-sh/bun/tree/7996d06b8f5869ccc9ad9a421cd06469da4273fc>)) on linux x86_64 [RunCommand]

Segmentation fault at address 0x00000008

- [`bundle_v2.zig:5611`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/bundler/bundle_v2.zig#L5611>): `src.bundler.bundle_v2.LinkerContext.findImportedFilesInCSSOrder`
- [`bundle_v2.zig:5010`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/bundler/bundle_v2.zig#L5010>): `src.bundler.bundle_v2.LinkerContext.computeChunks`
- [`bundle_v2.zig:4867`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/bundler/bundle_v2.zig#L4867>): `src.bundler.bundle_v2.LinkerContext.link`
- [`bundle_v2.zig:1908`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/bundler/bundle_v2.zig#L1908>): `src.bundler.bundle_v2.BundleV2.runFromJSInNewThread`
- [`bundle_v2.zig:2788`](<https://github.com/oven-sh/bun/blob/7996d06b8f5869ccc9ad9a421cd06469da4273fc/src/bundler/bundle_v2.zig#L2788>): `Thread.PosixThreadImpl.spawn__anon_186129.Instance.entryFn`

Features: Bun.stderr, Bun.stdout, jsc, shell, spawn, transpiler\_cache, tsconfig

### PR DESCRIPTION
**Pull request by zackradisic**
### What does this PR do?

Fixes #14487
Fixes #14420

Implement minifcation of these properties:
- [x] `-webkit-mask-composite`
- [x] `align-content`
- [x] `align-items`
- [x] `align-self`
- [x] `aspect-ratio`
- [x] `background-attachment`
- [x] `background-clip`
- [x] `background-color`
- [x] `background-image`
- [x] `background-origin`
- [x] `background-position-x`
- [x] `background-position-y`
- [x] `background-position`
- [x] `background-repeat`
- [x] `background-size`
- [x] `background`
- [x] `block-size`
- [x] `border-block-color`
- [x] `border-block-end-width`
- [x] `border-block-end`
- [x] `border-block-start-width`
- [x] `border-block-start`
- [x] `border-block-style`
- [x] `border-block-width`
- [x] `border-block`
- [x] `border-bottom-left-radius`
- [x] `border-bottom-right-radius`
- [x] `border-bottom-width`
- [x] `border-bottom`
- [x] `border-color`
- [x] `border-end-end-radius`
- [x] `border-end-start-radius`
- [x] `border-image-outset`
- [x] `border-image-repeat`
- [x] `border-image-slice`
- [x] `border-image-source`
- [x] `border-image-width`
- [x] `border-image`
- [x] `border-inline-color`
- [x] `border-inline-end-style`
- [x] `border-inline-end-width`
- [x] `border-inline-end`
- [x] `border-inline-start-style`
- [x] `border-inline-start-width`
- [x] `border-inline-start`
- [x] `border-inline-style`
- [x] `border-inline-width`
- [x] `border-inline`
- [x] `border-left-width`
- [x] `border-left`
- [x] `border-radius`
- [x] `border-right-width`
- [x] `border-right`
- [x] `border-start-end-radius`
- [x] `border-start-start-radius`
- [x] `border-style`
- [x] `border-top-left-radius`
- [x] `border-top-right-radius`
- [x] `border-top-width`
- [x] `border-top`
- [x] `border-width`
- [x] `border`
- [x] `box-align`
- [x] `box-direction`
- [x] `box-flex-group`
- [x] `box-flex`
- [x] `box-lines`
- [x] `box-ordinal-group`
- [x] `box-orient`
- [x] `box-pack`
- [x] `box-shadow`
- [x] `box-sizing`
- [x] `color`
- [x] `composes`
- [x] `column-gap`
- [x] `direction`
- [x] `display`
- [x] `flex-align`
- [x] `flex-basis`
- [x] `flex-direction`
- [x] `flex-flow`
- [x] `flex-grow`
- [x] `flex-item-align`
- [x] `flex-line-pack`
- [x] `flex-negative`
- [x] `flex-order`
- [x] `flex-pack`
- [x] `flex-positive`
- [x] `flex-preferred-size`
- [x] `flex-shrink`
- [x] `flex-wrap`
- [x] `flex`
- [x] `font-family`
- [x] `font-size`
- [x] `font-stretch`
- [x] `font-style`
- [x] `font-variant-caps`
- [x] `font-weight`
- [x] `height`
- [x] `inline-size`
- [x] `inset-block-end`
- [x] `inset-block-start`
- [x] `inset-block`
- [x] `inset-inline-end`
- [x] `inset-inline-start`
- [x] `inset-inline`
- [x] `inset`
- [x] `justify-content`
- [x] `justify-items`
- [x] `justify-self`
- [x] `line-height`
- [x] `margin-block-end`
- [x] `margin-block-start`
- [x] `margin-block`
- [x] `margin-bottom`
- [x] `margin-inline-end`
- [x] `margin-inline-start`
- [x] `margin-inline`
- [x] `margin-left`
- [x] `margin-right`
- [x] `margin-top`
- [x] `margin`
- [x] `mask-border-mode`
- [x] `mask-border-outset`
- [x] `mask-border-repeat`
- [x] `mask-border-slice`
- [x] `mask-border-source`
- [x] `mask-border-width`
- [x] `mask-border`
- [x] `mask-box-image-outset`
- [x] `mask-box-image-repeat`
- [x] `mask-box-image-slice`
- [x] `mask-box-image-source`
- [x] `mask-box-image-width`
- [x] `mask-box-image`
- [x] `mask-clip`
- [x] `mask-composite`
- [x] `mask-image`
- [x] `mask-mode`
- [x] `mask-origin`
- [x] `mask-position-x`
- [x] `mask-position-y`
- [x] `mask-position`
- [x] `mask-repeat`
- [x] `mask-size`
- [x] `mask-source-type`
- [x] `mask-type`
- [x] `mask`
- [x] `max-block-size`
- [x] `max-height`
- [x] `max-inline-size`
- [x] `max-width`
- [x] `min-block-size`
- [x] `min-height`
- [x] `min-inline-size`
- [x] `min-width`
- [x] `opacity`
- [x] `outline-color`
- [x] `outline-style`
- [x] `outline-width`
- [x] `outline`
- [x] `overflow-wrap`
- [x] `overflow-x`
- [x] `overflow-y`
- [x] `overflow`
- [x] `padding-block-end`
- [x] `padding-block-start`
- [x] `padding-block`
- [x] `padding-bottom`
- [x] `padding-inline-end`
- [x] `padding-inline-start`
- [x] `padding-inline`
- [x] `padding-left`
- [x] `padding-right`
- [x] `padding-top`
- [x] `padding`
- [x] `place-content`
- [x] `place-items`
- [x] `place-self`
- [x] `position`
- [x] `right`
- [x] `row-gap`
- [x] `scroll-margin-block-end`
- [x] `scroll-margin-block-start`
- [x] `scroll-margin-block`
- [x] `scroll-margin-bottom`
- [x] `scroll-margin-inline-end`
- [x] `scroll-margin-inline-start`
- [x] `scroll-margin-inline`
- [x] `scroll-margin-left`
- [x] `scroll-margin-right`
- [x] `scroll-margin-top`
- [x] `scroll-margin`
- [x] `scroll-padding-block-end`
- [x] `scroll-padding-block-start`
- [x] `scroll-padding-block`
- [x] `scroll-padding-bottom`
- [x] `scroll-padding-inline-end`
- [x] `scroll-padding-inline-start`
- [x] `scroll-padding-inline`
- [x] `scroll-padding-left`
- [x] `scroll-padding-right`
- [x] `scroll-padding-top`
- [x] `scroll-padding`
- [x] `text-decoration-color`
- [x] `text-emphasis-color`
- [x] `text-overflow`
- [x] `top`
- [x] `visibility`
- [x] `width`

### CODE REVIEW

**Comment by robobun** [comment]
@zackradisic, your commit da8d14f1173d92a64a68a4bab0fc5705e977b5d2 is building: [`#4689`](https://buildkite.com/bun/bun/builds/4689)

File: src/baby_list.zig [hunk 1]
```diff
@@ -14,6 +14,29 @@ pub fn BabyList(comptime Type: type) type {
...
+            };
+        }
+
+        pub fn toCss(this: *const ListType, comptime W: type, dest: *bun.css.Printer(W)) bun.css.PrintErr!void {
+            return bun.css.to_css.fromBabyList(Type, this, W, dest);
+        }
+
+        pub fn eql(lhs: *const ListType, rhs: *const ListType) bool {
+            if (lhs.len != rhs.len) return false;
+            for (lhs.sliceConst(), 0..) |*item, i| {
```
Comment (Jarred-Sumner) [inline src/baby_list.zig]:
```suggestion
            for (lhs.sliceConst(), rhs.sliceConst()) |*left, *right|  {
```

File: src/baby_list.zig [hunk 2]
```diff
@@ -14,6 +14,29 @@ pub fn BabyList(comptime Type: type) type {
...
+                if (!bun.css.generic.eql(Type, item, &rhs.ptr[i])) return false;
```
Comment (Jarred-Sumner) [inline src/baby_list.zig]:
```suggestion
                if (!bun.css.generic.eql(Type, item, right) return false;
```

File: src/baby_list.zig [hunk 3]
```diff
@@ -290,6 +313,11 @@ pub fn BabyList(comptime Type: type) type {
             return this.ptr[0..this.len];
         }
 
+        pub fn sliceConst(this: *const ListType) callconv(bun.callconv_inline) []const Type {
+            @setRuntimeSafety(false);
```
Comment (Jarred-Sumner) [inline src/baby_list.zig:317]:
```suggestion

```

File: src/bundler.zig [hunk 4]
```diff
@@ -950,21 +950,24 @@ pub const Bundler = struct {
...
-                        .result => |v| {
-                            const result = v.toCss(alloc, bun.css.PrinterOptions{
-                                .minify = bun.getenvTruthy("BUN_CSS_MINIFY"),
-                            }, null) catch |e| {
-                                bun.handleErrorReturnTrace(e, @errorReturnTrace());
-                                return null;
-                            };
-                            output_file.value = .{ .buffer = .{ .allocator = alloc, .bytes = result.code } };
-                        },
+                    var v = switch (bun.css.StyleSheet(bun.css.DefaultAtRule).parse(alloc, entry.contents, bun.css.ParserOptions.default(alloc, bundler.log), null)) {
```
Comment (Jarred-Sumner) [inline src/bundler.zig]:
```suggestion
                    var sheet = switch (bun.css.StyleSheet(bun.css.DefaultAtRule).parse(alloc, entry.contents, bun.css.ParserOptions.default(alloc, bundler.log), null)) {
```

File: src/css/css_parser.zig [hunk 5]
```diff
@@ -6016,166 +6135,170 @@ pub const serializer = struct {
...
+///
+/// This function only works on simple types like:
+/// - Simple equality types (e.g. integers, floats, strings, enums, etc.)
+/// - Types which implement a `.eql(lhs: *const @This(), rhs: *const @This()) bool` function
+///
+/// Or compound types composed of simple types such as:
+/// - Pointers to simple types
+/// - Optional simple types
+/// - Structs, Arrays, and Unions
+pub inline fn implementEql(comptime T: type, this: *const T, other: *const T) bool {
```
Comment (Jarred-Sumner) [inline src/css/css_parser.zig]:
```suggestion
pub fn implementEql(comptime T: type, this: *const T, other: *const T) bool {
```

File: src/css/css_parser.zig [hunk 6]
```diff
@@ -6016,166 +6135,170 @@ pub const serializer = struct {
...
+        const TT = std.meta.Child(T);
+        return implementEql(TT, this.*, other.*);
+    }
+    if (comptime @typeInfo(T) == .Optional) {
+        const TT = std.meta.Child(T);
+        if (this.* != null and other.* != null) return implementEql(TT, &this.*.?, &other.*.?);
+        return false;
+    }
+    return switch (tyinfo) {
+        .Optional => unreachable,
```
Comment (Jarred-Sumner) [inline src/css/css_parser.zig]:
```suggestion
        .Optional => @compileError("This function doesn't support optionals"),
```

File: src/css/css_parser.zig [hunk 7]
```diff
@@ -6016,166 +6135,170 @@ pub const serializer = struct {
...
+        .Pointer => unreachable,
```
Comment (Jarred-Sumner) [inline src/css/css_parser.zig]:
```suggestion
       .Pointer => @compileError("This function doesn't support pointers"),
```

File: src/css/selectors/selector.zig [hunk 8]
```diff
@@ -40,6 +47,430 @@ pub const impl = struct {
...
+        return c;
+    } else {
+        if (dir == .ltr) return Component{ .negation = langListToSelectors(allocator, RTL_LANGS) };
+        return Component{ .is = langListToSelectors(allocator, RTL_LANGS) };
+    }
+}
+
+fn langListToSelectors(allocator: Allocator, langs: []const []const u8) []Selector {
+    var selectors = allocator.alloc(Selector, langs.len) catch bun.outOfMemory();
+    for (langs, 0..) |lang, i| {
```
Comment (Jarred-Sumner) [inline src/css/selectors/selector.zig]:
```suggestion
    for (langs, selectors) |lang, *selector| {
```

File: src/css/selectors/selector.zig [hunk 9]
```diff
@@ -40,6 +47,430 @@ pub const impl = struct {
...
+        selectors[i] = Selector.fromComponent(allocator, Component{
```
Comment (Jarred-Sumner) [inline src/css/selectors/selector.zig]:
```suggestion
        selector.* = Selector.fromComponent(allocator, Component{
```

File: src/css/values/percentage.zig [hunk 10]
```diff
@@ -232,21 +228,189 @@ pub fn DimensionPercentage(comptime D: type) type {
...
+                else => this,
+            };
         }
 
         pub fn partialCmp(this: *const This, other: *const This) ?std.math.Order {
             _ = this; // autofix
             _ = other; // autofix
             @panic(css.todo_stuff.depth);
-        }
+        } // 128kb
```
Comment (Jarred-Sumner) [inline src/css/values/percentage.zig]:
```suggestion
        } 
```

**Pull request merged**